src/main.rs

```diff
@@ -1,35 +1,335 @@
+    // println!("Delay parameter: {}", delay_value);
+    // let gain_value = comb_filter.get_param(FilterParam::Gain);
+    // println!("Gain parameter: {}", gain_value);
+    // comb_filter.reset();
+    // let delay_value = comb_filter.get_param(FilterParam::Delay);
+    // println!("after reset:");
+    // println!("Delay parameter: {}", delay_value);
+    // let gain_value = comb_filter.get_param(FilterParam::Gain);
+    // println!("Gain parameter: {}", gain_value);
 
-    // Open the input wave file
+   
+
+    comb_filter.set_param(FilterParam::Delay, delay).unwrap();
+    comb_filter.set_param(FilterParam::Gain, gain).unwrap();
+    let delay_value = comb_filter.get_param(FilterParam::Delay);
+    println!("from args:");
+    println!("Delay parameter: {}", delay_value);
+    let gain_value = comb_filter.get_param(FilterParam::Gain);
+    println!("Gain parameter: {}", gain_value);
+
+
+    // // Open the input wave file
     let mut reader = hound::WavReader::open(&args[1]).unwrap();
     let spec = reader.spec();
-    let channels = spec.channels;
+    // let channels = spec.channels;
 
+    // Get WAV file properties
+    let channels = reader.spec().channels as usize;
+    let sample_rate = reader.spec().sample_rate as usize;
+    println!("{}, {}", channels, sample_rate);
+    
+    
     // TODO: Modify this to process audio in blocks using your comb filter and write the result to an audio file.
     //       Use the following block size:
-    let block_size = 1024;
+    let block_size : usize = 1024;
+    let mut pos = 0;
+    let mut data_length = 0;
+    for (i, sample) in reader.samples::<i16>().enumerate() {
+        let sample = sample.unwrap() as f32 / (1 << 15) as f32;
+        let cur_channel = i % channels;
+        unsafe{
+        data_array[cur_channel][pos] = sample;
+        }
+        if cur_channel == channels - 1
+        {
+            pos = pos + 1; 
+            data_length = pos;
+        }
+           
+    }
 
+    process_data_array(block_size, channels, &mut comb_filter, data_length);
+
+
+
+
+    
     // Read audio data and write it to the output text file (one column per channel)
     let mut out = File::create(&args[2]).expect("Unable to create file");
-    for (i, sample) in reader.samples::<i16>().enumerate() {
-        let sample = sample.unwrap() as f32 / (1 << 15) as f32;
-        write!(out, "{}{}", sample, if i % channels as usize == (channels - 1).into() { "\n" } else { " " }).unwrap();
+    for cur_data_index in 0..data_length
+    {
+        for cur_channel_index in 0..channels
+        {
+            let mut cur_data = 0.0;
+            unsafe
+            {
+                cur_data = data_array[cur_channel_index][cur_data_index];
+            }
+            if cur_channel_index == channels - 1
+            {
+                writeln!(out, "{}", cur_data).unwrap();
+            }
+            else 
+            {
+                write!(out, "{} ", cur_data).unwrap();
```

Comment:
I think we should write the data to an output wav file. You can check out this: https://docs.rs/hound/latest/hound/struct.WavWriter.html#